### PR TITLE
[DOCS] Update the PKI secrets engine docs title & description

### PR DESCRIPTION
**Pull request by yhyakuna**
### Description

This PR updates the secrets engine docs title & description for style consistency, as well as SEO improvement.

- Match the H1 header and the `page_title` for consistency
- Use `<Note>` callout elment instead of `~>`

### TODO only if you're a HashiCorp employee
- [x] **Backport Labels:** If this fix needs to be backported, use the appropriate `backport/` label that matches the desired release branch. Note that in the CE repo, the latest release branch will look like `backport/x.x.x`, but older release branches will be `backport/ent/x.x.x+ent`.
    - [ ] **LTS**: If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: website/content/docs/secrets/pki/index.mdx [hunk 1]
```diff
@@ -1,14 +1,22 @@
 ---
 layout: docs
-page_title: PKI - Secrets Engines
-description: The PKI secrets engine for Vault generates TLS certificates.
+page_title: PKI secrets engine
+description: Dynamically generate X.509 certificate with the PKI secrets engine plugin.
```
Comment (jonathanfrappier) [inline website/content/docs/secrets/pki/index.mdx:4]: `a certificate` or `certificates`? 
```suggestion
description: Dynamically generate X.509 certificates with the PKI secrets engine plugin.
```

**Review by jonathanfrappier — APPROVED** [review]

**Review by jonathanfrappier — APPROVED** [review]

File: website/content/docs/secrets/pki/considerations.mdx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 ---
 layout: docs
-page_title: 'PKI - Secrets Engines: Considerations'
-description: The PKI secrets engine for Vault generates TLS certificates.
+page_title: 'PKI secrets engine considerations'
+description: >-
+   Understand the important considerations and guidance before using the PKI secrets engine to generate the CA certificates.
```
Comment (victorr) [inline website/content/docs/secrets/pki/considerations.mdx:5]: Suggestion: change to "generate certificates", or even just change to end with "...before using the PKI secrets engine"

File: website/content/docs/secrets/pki/est.mdx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 ---
 layout: docs
-page_title: Enrollment over Secure Transport (EST) within Vault | PKI - Secrets Engines
-description: An overview of the Enrollment over Secure Transport protocol implementation within Vault.
+page_title: Enrollment over Secure Transport (EST)
+description: >-
+   Understand the configuration and limitations of Vault's PKI secrets engine implementation of the EST protocol.
```
Comment (victorr) [inline website/content/docs/secrets/pki/est.mdx:5]: Please consider spelling out "Enrollment over Secure Transport" in the description, to match the descriptions of the CIEPS and CMPv2 pages.

File: website/content/docs/secrets/pki/rotation-primitives.mdx [hunk 4]
```diff
@@ -1,7 +1,8 @@
 ---
 layout: docs
-page_title: 'PKI - Secrets Engine: Rotation Primitives'
-description: The PKI secrets engine for Vault generates TLS certificates.
+page_title: 'PKI secrets engine - rotation primitives'
+description: >-
+   Understand the X.509 certificate primitive fields when you are using the PKI secrets engine.
```
Comment (victorr) [inline website/content/docs/secrets/pki/rotation-primitives.mdx:5]: I don't think "primitive fields" is correct. Primitives here refer to certificate types, so maybe something along the lines of:
>Understand the primitive certificate types for root and intermediate CA rotation.

**Review by victorr — CHANGES_REQUESTED** [review]
I left a couple of suggestions but I am "requesting changes" as the description in file `rotation-primitives.mdx` needs a slight correction.

**Comment by yhyakuna** [comment]
@victorr Thank you so much for your help in updating the description! I believe I incorporated all your feedback, and it's ready for your review.

**Review by jonathanfrappier — APPROVED** [review]

**Review by victorr — APPROVED** [review]
Awesome. 👍

**Pull request merged**